src/components/main/index.tsx

```diff
@@ -17,36 +17,37 @@ import Loader from "../loader";
 type State = {
   movies: Movie[],
   moviesQuantity: number,
-  loading: boolean
+  reload: boolean,
+  loading: boolean,
+  typeSelected: string,
+  sortBySelected: string
 }
 
+const NUM_PAGE: string = "0";
+const CATEGORIES: string[] = ["all", "action", "documentary", "comedy", "horror", "crime"];
+
 function Main() {
   // Dispatch Instance
   const dispatch = useDispatch<AppDispatch>();
 
   // Redux State Extraction
-  const { movies, moviesQuantity, loading } = useSelector<RootState, State>((state) => state.movies);
-
-  // Local State
-  const [typeSelected, setTypeSelected] = React.useState("all");
-  const [sortBySelected, setSortBySelected] = React.useState("release_date");
+  const {
+    movies, moviesQuantity, reload, loading, typeSelected, sortBySelected,
+  } = useSelector<RootState, State>((state) => state.movies);
 
   React.useEffect(() => {
-    if (typeSelected === "all") {
-      dispatch(getAllMoviesAction("0", sortBySelected));
-    } else {
-      dispatch(getAllMoviesByGenreAction("0", typeSelected, sortBySelected));
+    if (reload) {
+      if (typeSelected === CATEGORIES[0]) {
+        dispatch(getAllMoviesAction(NUM_PAGE, sortBySelected));
```

Comment:
Also as i can see `getAllMoviesAction` and `getAllMoviesByGenreAction` almost the same functions. So the best option here will be to make one function and use `genre` as optional param. Smth like this:
```js
export function getAllMoviesByGenreAction(numPage: string,  sortBy: string, genre?: string) {
    return async(dispatch: AppDispatch) => {
        dispatch(getAllMoviesStart());

        try {
            const searchParams = new URLSearchParams({ limit: MOVIES_PER_PAGE, offset: numPage, sortBy, sortOrder: SORT_ORDER, filter: genre });
            const response = await fetch(`${ BACKEND_URL }?${searchParams}`);
            const data = await response.json();

            if (response.status === 200) {
                dispatch(getAllMoviesSuccess(data.data, data.totalAmount));
            }
        } catch (error) {
            dispatch(getAllMoviesError(error))
        }
    }
}
```

Reply:
Resolved, I need to declare two variables for params, cause if I pass to search params filter with "" or null, the API doesn't get a response